bdk_electrum/src/lib.rs

```diff
@@ -0,0 +1,583 @@
+                tx.clone()
+            } else {
+                let res_tx = client.transaction_get(&res.tx_hash)?;
+                if !res_tx
+                    .input
+                    .iter()
+                    .any(|txin| txin.previous_output == outpoint)
+                {
+                    continue;
+                }
+                has_spending = true;
+                res_tx
+            };
+            full_txs.insert(res.tx_hash, res_tx);
+
+            let tx_height = match res.height {
+                // Electrum server API specifies that height <= 0 is considered unconfirmed
+                h if h <= 0 => {
+                    debug_assert!(
+                        h == 0 || h == -1,
+                        "unexpected height ({}) from electrum server",
+                        h
+                    );
+                    TxHeight::Unconfirmed
+                }
+                h => {
+                    let h = h as u32;
+                    if h > tip.height {
+                        TxHeight::Unconfirmed
+                    } else {
+                        TxHeight::Confirmed(h)
+                    }
+                }
+            };
+
+            if let Err(failure) = update.insert_tx(txid, tx_height) {
+                match failure {
+                    sparse_chain::InsertTxError::TxTooHigh { .. } => {
+                        unreachable!("we should never encounter this as we ensured height <= tip");
+                    }
+                    sparse_chain::InsertTxError::TxMovedUnexpectedly { .. } => {
+                        return Err(InternalError::Reorg);
+                    }
+                }
+            }
+        }
+    }
+    Ok(full_txs)
+}
+
+/// Populate an update [`SparseChain`] with transactions (and associated block positions) from
+/// the given `txids`.
+fn populate_with_txids(
+    client: &Client,
+    update: &mut SparseChain,
+    txids: &mut impl Iterator<Item = Txid>,
+) -> Result<(), InternalError> {
+    let tip = update
+        .latest_checkpoint()
+        .expect("update must have atleast one checkpoint");
+    for txid in txids {
+        let tx = match client.transaction_get(&txid) {
+            Ok(tx) => tx,
+            Err(electrum_client::Error::Protocol(_)) => continue,
+            Err(other_err) => return Err(other_err.into()),
+        };
+
+        let spk = tx
+            .output
+            .get(0)
+            .map(|txo| &txo.script_pubkey)
+            .expect("tx must have an output");
+
+        let tx_height = match client
+            .script_get_history(spk)?
+            .into_iter()
+            .find(|r| r.tx_hash == txid)
+            .map(|r| r.height)
+        {
+            Some(h) if h > 0 => TxHeight::Confirmed(h as _),
+            Some(_) => TxHeight::Unconfirmed,
+            None => continue,
+        };
+
+        if tx_height.is_confirmed() && tx_height > TxHeight::Confirmed(tip.height) {
+            continue;
+        }
```

Comment:
Shouldn't you insert tx as unconfirmed in this case, as you do above, in populate_outpoints?

Reply:
Yes. Thanks again

Reply:
Fixed.